modules/dhcp_isc/isc_state_changes_observer.rb

```diff
@@ -166,9 +166,10 @@ def update_subnet_service_with_dhcp_records(records)
               if dupe = service.find_host_by_mac(record.subnet_address, record.mac)
                 service.delete_host(dupe)
               end
-              if dupe = service.find_host_by_ip(record.subnet_address, record.ip)
-                service.delete_host(dupe)
-              end
+              #TODO: What is the aim of this?
```

Comment:
This bit checks if subnet service already has host records with the same mac or ip address, and removes them. I haven't checked, but I assume this is how isc dhcpd handles such conflicts.